hw/mcu/nordic/nrf52xxx/src/hal_timer.c

```diff
@@ -168,11 +168,15 @@ nrf_timer_set_ocmp(struct nrf52_hal_timer *bsptimer, uint32_t expiry)
          * The nrf documentation states that you must set the output
          * compare to 2 greater than the counter to guarantee an interrupt.
          * Since the counter can tick once while we check, we make sure
-         * it is greater than 2.
+         * it is greater than 2. Otherwise enable TICK event so we can get an
+         * interrupt at proper tick (also we may get an extra interrupt, but
+         * we will just not handle any timer there).
          */
         if (delta_t < 3) {
```

Comment:
so if I am understanding this correctly, this code will keep interrupting with the tick timer until we reach the expiration time. Given that is the case, I am wondering if we should fire off the timer if delta_t is zero. Otherwise we could end up being a tick late (at least that seems possible).

Reply:
you're right. and now actually I think we should also set pending interrupt when delta==1 because the timer can tick once so TICK interrupt will happen on next tick which is also one tick too late. so something like this should work I think:
delta==2 enable tick and wait for proper tick (we may have 1 or 2 additional "empty" interrupts)
delta==1 enable tick and set irq pending (we may have 0 or 1 "empty" interrupt)
delta==0 set irq pending

does this make sense?

Reply:
yes that seems to make sense to me. Too bad it cannot just work on the exact tick :-)